python/ray/air/util/tensor_extensions/arrow.py

```diff
@@ -520,8 +528,16 @@ def from_numpy(
 
         # Whether all subndarrays are contiguous views of the same ndarray.
         shapes, sizes, raveled = [], [], []
+        ndim = None
         for a in arr:
             a = np.asarray(a)
+            if ndim is not None and a.ndim != ndim:
+                raise ValueError(
+                    "ArrowVariableShapedTensorArray only supports tensor elements that "
+                    "all have the same number of dimensinos, but got tensor elements "
```

Comment:
nit: "dimensinos" seems not fixed yet

Reply:
Oh weird, I thought I fixed that, good catch!